### ISSUE TITLE
Hard to repro failed push with large files and lfs-test-server

### ISSUE THREAD

**Issue by aggieNick02** [git-lfs/git-lfs#3416]
We have a project that needs to track large (>4GB) files so we spun up a little lfs-test-server on ubuntu18.04 LTS to go along with our repo on github.com. We're on the latest lfs-test-server (built from source) and latest git for windows. We started testing some things out with a dummy repo.

We're using the workaround of skipping smudge to work with these files on windows. It seems to work pretty well most of the time. But a few minutes ago I ran `git push` after a commit with a new big file, and got this output:

> `Uploading LFS objects:   0% (0/1), 9.1 GB | 104 MB/s, done
> Fatal error: Server error: http://10.100.0.1:9999/objects/2400552958e23519156f20f3256b49bc572cd9d728ef50a8d93702d886813ab8
> error: failed to push some refs to 'git@github.com:aggieNick02/hello-world.git`

I checked the lfs-test-server's file store and the entire new file, with correct contents, was present (sha256 matched). The log for lfs-test-server showed no errors:

> `2018-12-06T20:18:08Z bmaster-dev lfs[17919] [server.go:342]: method=POST url=/objects/batch status=200 request_id=7448c73b-a8c8-e706-004b-f0c387c7b411
> 2018-12-06T20:18:08Z bmaster-dev lfs[17919] [server.go:342]: method=POST url=/objects/batch status=200 request_id=8324022a-491a-ab35-91ef-696d3b650979
> 2018-12-06T20:18:08Z bmaster-dev lfs[17919] [server.go:342]: url=/objects/batch status=200 request_id=f1cb516f-58bc-da0c-6c9c-18cd50faed95 method=POST
> 2018-12-06T20:18:08Z bmaster-dev lfs[17919] [server.go:342]: method=POST url=/objects/batch status=200 request_id=61caa3ad-87f8-4322-d697-8dcf248e71bd
> 2018-12-06T20:18:08Z bmaster-dev lfs[17919] [server.go:342]: method=POST url=/objects/batch status=200 request_id=053b138a-4d0f-a740-5968-bfa3391d7687
> 2018-12-06T20:18:08Z bmaster-dev lfs[17919] [server.go:342]: request_id=68dac289-c26f-41db-76b0-6c18f6ad0d65 method=POST url=/objects/batch status=200
> 2018-12-06T20:18:08Z bmaster-dev lfs[17919] [server.go:342]: method=POST url=/objects/batch status=200 request_id=f37540e9-2710-b812-43c8-eda61ebfac91
> 2018-12-06T20:18:08Z bmaster-dev lfs[17919] [server.go:342]: method=POST url=/objects/batch status=200 request_id=fc16bd86-5627-0d33-1309-389fef6457af
> 2018-12-06T20:18:11Z bmaster-dev lfs[17919] [server.go:361]: url=/objects/2400552958e23519156f20f3256b49bc572cd9d728ef50a8d93702d886813ab8 status=200 request_id=67556744-5757-00ab-aaf9-22de77eff880 method=PUT`

At a loss, I ran `git push` again. This time, success. (The only thing worse than failure, because now it's an intermittent issue. ;-) ) Also bizarre was that the output showed uploading twice the size of the new file:

> `$ git push
> Uploading LFS objects: 100% (1/1), 18 GB | 104 MB/s, done
> Enumerating objects: 6, done.
> Counting objects: 100% (6/6), done.
> Delta compression using up to 8 threads
> Compressing objects: 100% (4/4), done.
> Writing objects: 100% (4/4), 514 bytes | 257.00 KiB/s, done.
> Total 4 (delta 1), reused 0 (delta 0)
> remote: Resolving deltas: 100% (1/1), completed with 1 local object.
> To github.com:aggieNick02/hello-world.git
>    de87404..b617ba8  master -> master`

Any ideas? Is there anything I can enable so that the next time this happens I can get more info about what is failing?

**Comment by bk2204** [comment]
Yeah, as you've noticed, the "Fatal error: Server error:" response indicates a 500 error. I'm not sure why the test server is producing that in this case, but it does seem to be an issue there. I'm afraid that the test server doesn't get a huge amount of attention, but that could be either because it generally works or nobody's using it, I'm not sure which.

I'm going to transfer this issue over to that repository.
